presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestDetermineJoinDistributionType.java

```diff
@@ -62,7 +63,7 @@
     @BeforeClass
     public void setUp()
     {
-        tester = new RuleTester(ImmutableList.of(), ImmutableMap.of(), Optional.of(NODES_COUNT));
+        tester = defaultRuleTester(ImmutableList.of(), ImmutableMap.of(), Optional.of(NODES_COUNT));
```

Comment:
Maybe you could introduce a builder for `RuleTester` instead of factory method that takes empty arguments

Reply:
the most critical object for constructing a `RuleTester` is the `queryRunner`, so I thought that using a factory to create a queryRunner when not provided should be appropriate. If we use a constructor instead, we end up initializing the member variables in multiple constructors.